lib/debugger/debugger.dart

```diff
@@ -0,0 +1,1309 @@
+        });
+      }
+
+      if (!breakpoint.resolved) {
+        element.add(span(text: ' (unresolved)', c: 'subtle'));
+      }
+
+      return element;
+    });
+  }
+
+  Stream<Breakpoint> get onDoubleClick => _items.onDoubleClick;
+
+  final CoreElement _breakpointsCountDiv;
+
+  SelectableList<Breakpoint> _items;
+
+  CoreElement get element => _items;
+
+  Stream<Breakpoint> get onSelectionChanged => _items.onSelectionChanged;
+
+  void showBreakpoints(List<Breakpoint> breakpoints) {
+    breakpoints = breakpoints.toList();
+    breakpoints.sort(_breakpointComparator);
+
+    _items.setItems(breakpoints);
+    _breakpointsCountDiv.text = breakpoints.length.toString();
+  }
+}
+
+class ScriptsView {
+  ScriptsView(URIDescriber uriDescriber) {
+    _items = new SelectableList<ScriptRef>()
+      ..flex()
+      ..element.style.overflowY = 'scroll';
+    _items.setRenderer((ScriptRef scriptRef) {
+      final String uri = scriptRef.uri;
+      final String name = uriDescriber(uri);
+      final CoreElement element = li(text: name, c: 'list-item');
+      if (name != uri) {
+        element.add(span(text: ' $uri', c: 'subtle'));
+      }
+      element.tooltip = uri;
+      return element;
+    });
+  }
+
+  SelectableList<ScriptRef> _items;
+
+  String rootLib;
+
+  List<ScriptRef> get items => _items.items;
+
+  CoreElement get element => _items;
+
+  Stream<ScriptRef> get onSelectionChanged => _items.onSelectionChanged;
+
+  void showScripts(
+    List<ScriptRef> scripts,
+    String rootLib,
+    String commonPrefix, {
+    bool selectRootScript = false,
+  }) {
+    this.rootLib = rootLib;
+
+    scripts.sort((ScriptRef ref1, ScriptRef ref2) {
+      String uri1 = ref1.uri;
+      String uri2 = ref2.uri;
+
+      if (uri1.startsWith('dart:_')) {
+        uri1 = uri1.replaceAll('dart:_', 'dart:');
+      }
+      if (uri2.startsWith('dart:_')) {
+        uri2 = uri2.replaceAll('dart:_', 'dart:');
+      }
+
+      if (commonPrefix != null) {
+        if (uri1.startsWith(commonPrefix) && !uri2.startsWith(commonPrefix)) {
+          return -1;
+        } else if (!uri1.startsWith(commonPrefix) &&
+            uri2.startsWith(commonPrefix)) {
+          return 1;
+        }
+      }
+
+      if (uri1.startsWith('dart:') && !uri2.startsWith('dart:')) {
+        return 1;
+      } else if (!uri1.startsWith('dart:') && uri2.startsWith('dart:')) {
+        return -1;
+      }
+
+      return uri1.compareTo(uri2);
+    });
+
+    ScriptRef selection;
+    if (selectRootScript) {
+      selection = scripts.firstWhere((script) => script.uri == rootLib,
+          orElse: () => null);
+    }
+    _items.setItems(scripts, selection: selection);
+  }
+
+  void clearScripts() => _items.clearItems();
+}
+
+class CallStackView {
+  CallStackView() {
+    _items = new SelectableList<Frame>()
+      ..flex()
+      ..clazz('menu-item-bottom-border')
+      ..element.style.overflowY = 'scroll';
```

Comment:
looks like all the menus  allow scroll. maybe add that to the css file?

Reply:
done